[Canvas] Updates function reference docs (#64741)
elastic/kibana: docs/canvas/canvas-function-reference.asciidoc

----
`as` casts any primitive value (`string`, `number`, `date`, `null`) into a `datatable` with a single row and a single column with the given name (or defaults to `"value"` if no name is provided). This is useful when piping a primitive value into a function that only takes `datatable` as an input.

> In the example above, `ply` expects each `fn` subexpression to return a `datatable` in order to merge the results of each `fn` back into a `datatable`, but using a `math` aggregation in the subexpressions returns a single `math` value, which is then cast into a `datatable` using `as`.

Reviewer: To meet accessibility standards, we are no longer able to use words like `above`, `below`, `left`, or `right` in our docs.

Query Elasticsearch for raw documents. Specify the fields you want to retrieve, especially if you are asking for a lot of rows.

*Expression syntax*
[source,js]
----
esdocs index="logstash-*"
esdocs "currency:"EUR"" index="kibana_sample_data_ecommerce"
esdocs query="response:404" index="kibana_sample_data_logs"
esdocs index="kibana_sample_data_flights" count=100
esdocs index="kibana_sample_data_flights" sort="AvgTicketPrice, asc"
----

*Code example*
[source,text]
----
filters
| esdocs index="kibana_sample_data_ecommerce"
  fields="customer_gender, taxful_total_price, order_date"
  sort="order_date, asc"
  count=10000
| mapColumn "order_date"
  fn={getCell "order_date" | date {context} | rounddate "YYYY-MM-DD"}
| alterColumn "order_date" type="date"
| pointseries x="order_date" y="sum(taxful_total_price)" color="customer_gender"
| plot defaultStyle={seriesStyle lines=3}
  palette={palette "#7ECAE3" "#003A4D" gradient=true}
| render
----

> This retrieves the first 10000 documents data from the `kibana_sample_data_ecommerce` index sorted by `order_date` in ascending order and only requests the `customer_gender`, `taxful_total_price`, and `order_date` fields.

Reviewer: ```suggestion
This retrieves the first 10000 documents data from the `kibana_sample_data_ecommerce` index sorted by `order_date` in ascending order, and only requests the `customer_gender`, `taxful_total_price`, and `order_date` fields.
```

[[formatnumber_fn]]
=== `formatnumber`

> Formats a number into a formatted number string using Numeral pattern.

Reviewer: ```suggestion
Formats a number into a formatted number string using the Numeral pattern.
```

[[m_fns]]
== M

[float]
[[mapCenter_fn]]
=== `mapCenter`

> Returns an object with the center coordinates and zoom level of the map

Reviewer: ```suggestion
Returns an object with the center coordinates and zoom level of the map.
```

*Accepts:* `null`

[cols="3*^<"]
|===
|Argument |Type |Description

|`lat` ***
|`number`
|Latitude for the center of the map

|`lon` ***
|`number`
|Longitude for the center of the map

|`zoom` ***
|`number`

> |The zoom level of the map

Reviewer: ```suggestion
|Zoom level of the map
```

|The CSS font properties for the content. For example, "font-family" or "font-weight".

Default: `${font}`

|`openLinksInNewTab`
|`boolean`

> |A true/false value for opening links in a new tab. Default value is false. Setting to true will open all links in a new tab.

Reviewer: ```suggestion
|A true or false value for opening links in a new tab. the default value is `false`. Setting to `true` opens all links in a new tab.
```

[[s_fns]]
== S

[float]
[[savedLens_fn]]
=== `savedLens`

> Returns an embeddable for a saved lens object

Reviewer: ```suggestion
Returns an embeddable for a saved Lens visualization object.
```

*Accepts:* `any`

[cols="3*^<"]
|===
|Argument |Type |Description

|`id`
|`string`

> |The ID of the Saved Lens Object

Reviewer: ```suggestion
|The ID of the saved Lens visualization object
```

|`timerange`
|`timerange`
|The timerange of data that should be included

|`title`
|`string`

> |The title for the lens emebeddable

Reviewer: ```suggestion
|The title for the Lens visualization object
```

|===

*Returns:* `embeddable`

[float]
[[savedMap_fn]]
=== `savedMap`

> Returns an embeddable for a saved map object

Author: ```suggestion
Returns an embeddable for a saved map object.
```

*Accepts:* `any`

[cols="3*^<"]
|===
|Argument |Type |Description

|`center`
|`mapCenter`
|The center and zoom level the map should have

|`hideLayer` †
|`string`
|The IDs of map layers that should be hidden

|`id`
|`string`

> |The ID of the Saved Map Object

Reviewer: ```suggestion
|The ID of the saved map object
```

|`timerange`
|`timerange`
|The timerange of data that should be included

|`title`
|`string`
|The title for the map
|===

*Returns:* `embeddable`

[float]
[[savedVisualization_fn]]
=== `savedVisualization`

> Returns an embeddable for a saved visualization object

Reviewer: ```suggestion
Returns an embeddable for a saved visualization object.
```

*Accepts:* `any`

[cols="3*^<"]
|===
|Argument |Type |Description

|`colors` †
|`seriesStyle`

> |Define the color to use for a specific series

Reviewer: ```suggestion
|Defines the color to use for a specific series
```

|`hideLegend`
|`boolean`

> |Should the legend be hidden

Reviewer: ```suggestion
|Specifies the option to hide the legend
```

|`id`
|`string`

> |The ID of the Saved Visualization Object

Reviewer: ```suggestion
|The ID of the saved visualization object
```

*Returns:* `datatable`

[float]
[[timerange_fn]]
=== `timerange`

> An object that represents a span of time

Reviewer: ```suggestion
An object that represents a span of time.
```
